### PR TITLE
Update chips with new theming support

### PR DESCRIPTION
**Pull request by chillkang**
## Done
- update chips with light and dark theme support
- add examples for dark theme version

## Fixes 
- https://warthogs.atlassian.net/browse/WD-7600

## QA
- Open [demo](https://vanilla-framework-4986.demos.haus/docs/examples/patterns/chip/variants-dark) and compare with [the existing version](https://vanillaframework.io/docs/examples/patterns/chip/variants) to check if chips components behave as expected

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots
Screenshot 2024-02-08 at 6 16 44 PM

### CODE REVIEW

File: templates/docs/examples/patterns/chip/dark.html [hunk 1]
```diff
@@ -3,28 +3,29 @@
 
 {% block standalone_css %}patterns_chip{% endblock %}
 
+{% set is_dark = true %}
```
Comment (bartaz) [inline templates/docs/examples/patterns/chip/dark.html:6]: I know that this way would be more consistent with other components we are migrating, but we already have the variants example with both light and dark theme, and I think it's OK to leave it as is, so we can also easier compare if the changes work as expected.

So please revert the variants example back to having both light and dark chips, thanks!

File: scss/_patterns_chip.scss [hunk 2]
```diff
@@ -64,234 +65,204 @@ $chip-border-thickness: 1px;
...
-  @include vf-chip-theme(
-    $chip-type: $chip-type,
-    $is-interactive: $is-interactive,
-    $colors-chip-tinted-backgrounds: $colors-dark-theme--tinted-backgrounds,
-    $colors-chip-tinted-borders: $colors-dark-theme--tinted-borders,
-    $color-chip-value: $colors--dark-theme--text-default,
-    $color-chip-lead: $colors--dark-theme--text-default
-  );
+  // stylelint-disable-next-line selector-max-type
+  body.is-dark .p-chip--information .p-chip__dismiss {
```
Comment (bartaz) [inline scss/_patterns_chip.scss:265]: The dismiss button doesn't seem to depend on the variant of the chip, only on the theme. It's the same in all the variants (within the same theme), so seems redundant to have it defined for all of them separately: `body.is-dark .p-chip--information .p-chip__dismiss` `body.is-dark .p-chip--caution .p-chip__dismiss`, etc.

Would it work if defined in one place as `body.is-dark .p-chip__dismiss` for all the chips?

Comment (bartaz) [reply scss/_patterns_chip.scss:265]: Also, currently it only work with `is-dark` set on body, but we need it to work with `is-dark` set on chip itself.

So it needs to be something like:

```
body.is-dark .p-chip__dismiss,
.p-chip.is-dark .p-chip__dismiss,
[class*='.p-chip--'].is-dark .p-chip__dismiss { // this [class*=] will match all class names containing `p-chip--` which should be all our chips variants
    @include vf-icon-close(#fff);
}

**Comment by lyubomir-popov** [comment]
can we please use this as the new yellow on the light theme - it is further from red and accessible CC7900

**Review by bartaz — APPROVED** [review]
Looks great, thanks for completing this!

**Pull request merged**